### PR TITLE
Admin bar: update to match with Nav Redesign i3

### PR DESCRIPTION
**Pull request by fushar**
Related to:

- https://github.com/Automattic/dotcom-forge/issues/8144

## Proposed changes:

This PR updates the admin bar for WordPress.com sites as follows:

1. Repurposes the WordPress logo as a link to /sites.
2. Adds Reader menu.
3. Hides the updates icon (via CSS).
4. Hides the comments icon (via CSS).

---

I also took the opportunity to split the `wpcom-site-menu` feature (which is getting too large) into 3:

- `wpcom-admin-bar`
- `wpcom-admin-menu`
- `wpcom-sidebar-notice`

With this, the responsibility of each feature is clear and localized.

|wp-admin|
|-|
|Before<br>image|
|After<br>image|

|site front-end|
|-|
|Before<br>image|
|After<br>image|

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

1. Patch this to WordPress.com Features on Jetpack Beta Tester.
3. Smoke-test wp-admin and site frontend and verify the screenshot.
4. Also test on Simple/Atomic Classic/Default sites (4 combinations).
5. (Regression testing) Test that you can still view the sidebar upsell in all cases.

### CODE REVIEW

File: projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.php [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	);
+}
+add_action( 'admin_bar_menu', 'wpcom_repurpose_wp_logo_as_all_sites_menu', 11 );
+
+/**
+ * Adds the Reader menu.
+ *
+ * @param WP_Admin_Bar $wp_admin_bar The WP_Admin_Bar core object.
+ */
+function wpcom_add_reader_menu( $wp_admin_bar ) {
```
Comment (okmttdhr) [inline projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.php:67]: badge

Should we move the Reader menu to the right most according to the design in pfsHM7-1kz-p2 or https://github.com/Automattic/dotcom-forge/issues/8144? Did I miss something? 

this PR
Screenshot 2024-07-09 at 11 09 03

Figma
Screenshot 2024-07-09 at 11 10 37

Comment (fushar) [reply projects/packages/jetpack-mu-wpcom/src/features/wpcom-admin-bar/wpcom-admin-bar.php:67]: Good question! We had a p1720413874823569/1720204941.135799-slack-C06DN6QQVAQ about this on Slack, and @davemart-in approved this:

> > Would it be worth putting Reader on the left like this? It's a bit more global than the site level and means it won't move around when navigating in and out of site contexts.
> 
> Yes. Let's do that!

**Review by okmttdhr — DISMISSED** [review]
Worked well for me 👍 

- ✅Simple/Atomic Classic (changes applied)
- ✅Simple/Atomic Default (no regression)

We'll address the items in the Site Name in a follow-up PR, correct? 

| Figma | Atomic | Simple |
|--------|--------|--------|
| Screenshot 2024-07-09 at 11 37 12 We might want to add "Dashboard" | Screenshot 2024-07-09 at 11 38 26 | Screenshot 2024-07-09 at 11 38 34 (Manage Site may be super-admin only) |

**Comment by fushar** [comment]
> We'll address the items in the Site Name in a follow-up PR, correct?

We had a discussion about this exact thing in the issue https://github.com/Automattic/dotcom-forge/issues/8144#issuecomment-2213263720; we're gonna hold off adding that on wp-admin admin bar (for now).

**Comment by fushar** [comment]
@okmttdhr, I just rebased and loses your review; could you quickly help smoke-test it again and hit the approve button again 🙏

**Review by okmttdhr — APPROVED** [review]
LGTM with a test on Atomic Classic

**Pull request merged**